plugins/inputs/ethtool/ethtool_linux.go

```diff
@@ -80,19 +77,27 @@ func (e *Ethtool) gatherEthtoolStats(iface net.Interface, acc telegraf.Accumulat
 		return
 	}
 
+	fields[fieldInterfaceUp] = e.interfaceUp(iface)
 	for k, v := range stats {
 		fields[k] = v
 	}
 
 	acc.AddFields(pluginName, fields, tags)
 }
 
+func (e *Ethtool) interfaceUp(iface net.Interface) int {
+	interfaceUp := 0
+	if iface.Flags&net.FlagUp != 0 {
+		interfaceUp = 1
+	}
+	return interfaceUp
+}
+
```

Comment:
Isn't a boolean more appropriate given that you have only two states?
```suggestion
func (e *Ethtool) interfaceUp(iface net.Interface) bool {
	return (iface.Flags&net.FlagUp) != 0
}

```

Reply:
I also think this is more logical. What do you think @zak-pawel?